### PR TITLE
[KRV-24554] Auth proxy e2e

### PR DESCRIPTION
**Pull request by EvgenyUglov**
# Description

1. Add Install/Delete authorization CRDs steps to the test scenarios where applicable
2. Use dellctl to create authorization CRs in authorization proxy e2e tests
3. Add Storage/CSMRole/CSMTenant test CRs for e2e

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/1281|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Ran authorization-proxy-server e2e tests to deploy auth module, install CRDs, create and configure the appropriate CRs to make sure these steps are successfully executed
```
  STEP: Getting test environment variables @ 06/10/24 22:05:30.26
  STEP: [authorizationproxyserver] @ 06/10/24 22:05:30.26
  STEP: Reading values file @ 06/10/24 22:05:30.26
  STEP: Getting a k8s client @ 06/10/24 22:05:30.279
[BeforeSuite] PASSED [0.057 seconds]
------------------------------
[run-e2e-test] E2E Testing Running all test Given Test Scenarios
/home/evgeny/workspace/csm/csm-operator/tests/e2e/e2e_test.go:123
  STEP: Starting: Install Authorization Proxy Server  @ 06/10/24 22:05:30.317
  STEP:      Executing  Given an environment with k8s or openshift, and CSM operator installed @ 06/10/24 22:05:30.317
  STEP:      Executing  Install Authorization CRDs [2] @ 06/10/24 22:05:30.357
  STEP:      Executing  Create [authorization-proxy-server] prerequisites from CR [1] @ 06/10/24 22:05:32.189
=== Creating Authorization Proxy Server Prerequisites ===
  STEP:      Executing  Apply custom resource [1] @ 06/10/24 22:05:44.757
  Jun 10 22:05:44.761: INFO: Running '/usr/local/bin/kubectl --namespace=authorization apply --validate=true -f -'
  Jun 10 22:05:45.355: INFO: stderr: ""
  Jun 10 22:05:45.355: INFO: stdout: "containerstoragemodule.storage.dell.com/authorization created\nconfigmap/csm-config-params created\n"
  STEP:      Executing  Validate [authorization-proxy-server] module from CR [1] is installed @ 06/10/24 22:05:45.356
  STEP:      Executing  Configure authorization-proxy-server for [powerscale] for CR [1] @ 06/10/24 22:09:16.408
=== Configuring Authorization Proxy Server ===
=== Generating Admin Token ===
=== Writing Admin Token to Tmp File ===
 
=== Creating Storage ===
 
=== Storage === 
 /usr/local/bin/kubectl apply -f testfiles/authorization-templates/csm-authorization_v1_storage.yaml

=== Creating Tenant ===
 
=== Tenant === 
 /usr/local/bin/kubectl apply -f testfiles/authorization-templates/csm-authorization_v1_csmtenant.yaml

=== Creating Role ===

=== Role === 
 /usr/local/bin/kubectl apply -f testfiles/authorization-templates/csm-authorization_v1_csmrole.yaml

=== Generating token ===
 
=== Token ===
 /usr/local/bin/dellctl generate token --admin-token /tmp/adminToken.yaml --access-token-expiration 10m0s --refresh-token-expiration 48h --tenant csmtenant-sample --insecure --addr csm-authorization.com:31797

=== Applying token ===
 
=== Token Applied ===
 
  STEP:      Executing  Configure authorization-proxy-server for [powerflex] for CR [1] @ 06/10/24 22:12:17.076
=== Configuring Authorization Proxy Server ===
=== Generating Admin Token ===
=== Writing Admin Token to Tmp File ===
 
=== Creating Storage ===
 
=== Storage === 
 /usr/local/bin/kubectl apply -f testfiles/authorization-templates/csm-authorization_v1_storage.yaml

=== Creating Tenant ===
 
=== Tenant === 
 /usr/local/bin/kubectl apply -f testfiles/authorization-templates/csm-authorization_v1_csmtenant.yaml

=== Creating Role ===

=== Role === 
 /usr/local/bin/kubectl apply -f testfiles/authorization-templates/csm-authorization_v1_csmrole.yaml

=== Generating token ===
 
=== Token ===
 /usr/local/bin/dellctl generate token --admin-token /tmp/adminToken.yaml --access-token-expiration 10m0s --refresh-token-expiration 48h --tenant csmtenant-sample --insecure --addr csm-authorization.com:31797

=== Applying token ===
 
=== Token Applied ===
 
  STEP:      Executing  Delete custom resource [1] @ 06/10/24 22:12:27.479
  STEP:      Executing  Delete Authorization CRDs [2] @ 06/10/24 22:12:27.543
  STEP: Ending: Install Authorization Proxy Server
   @ 06/10/24 22:12:33.474
• [428.160 seconds]
------------------------------

Ran 1 of 1 Specs in 428.217 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS

Ginkgo ran 1 suite in 10m3.606309195s
Test Suite Passed
```

### CODE REVIEW

File: tests/e2e/testfiles/authorization-templates/csm-authorization_v1_storage.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: csm-authorization.storage.dell.com/v1alpha1
+kind: Storage
+metadata:
+  name: powerscale
+spec:
+  type: powerscale
+  endpoint: https://10.230.24.241:8080
```
Comment (atye) [inline tests/e2e/testfiles/authorization-templates/csm-authorization_v1_storage.yaml:7]: The spec values in the CRs should be updated during the test so users, and us, can pass in different values. This internal IP should be removed.

We can have uses put values in https://github.com/dell/csm-operator/blob/main/tests/e2e/array-info.sh and put them into the CRs.

**Review by alikdell — DISMISSED** [review]

**Review by shanmydell — CHANGES_REQUESTED** [review]
Can we update go.mod to use the latest k8s so that dependabot alert gets resolved

File: tests/e2e/testfiles/values.yaml [hunk 2]
```diff
@@ -2,16 +2,19 @@
 - scenario: "Install Authorization Proxy Server"
   paths:
     - "testfiles/authorization-templates/csm_authorization_proxy_server.yaml"
+    - "testfiles/authorization-templates/csm_authorization_crds.yaml"
   modules:
     - "authorizationproxyserver"
   steps:
     - "Given an environment with k8s or openshift, and CSM operator installed"
+    - "Install Authorization CRDs [2]"
```
[the remark this answers is not in the document]
Comment (EvgenyUglov) [reply tests/e2e/testfiles/values.yaml:10]: New "Install/Delete CRDs" steps are already added there in `authorization-gitops` branch - https://github.com/dell/csm-operator/blob/abab27b164e22081866e9b3d518ae482f3c03fcc/tests/e2e/steps/steps_runner.go#L84

File: tests/e2e/testfiles/authorization-templates/csm-authorization_v1_storage.yaml (hunk 1 above)
Comment (EvgenyUglov) [reply tests/e2e/testfiles/authorization-templates/csm-authorization_v1_storage.yaml]: Right, missed that, thanks! Fixed.
Yeah, we can add populate them from that script, but I'd leave this change for another PR

File: tests/e2e/testfiles/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (shaynafinocchiaro) [reply tests/e2e/testfiles/values.yaml:10]: I wasn't on the right branch when I looked, sorry about that!

**Review by alikdell — APPROVED** [review]

**Review by KerryKovacevic — APPROVED** [review]

**Pull request merged**